pkg/cmd/openshift-operators/apis/operators/v1alpha1/types.go

```diff
@@ -0,0 +1,86 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ManagementState string
+
+const (
+	// Enabled means that the operator is actively managing its resources and trying to keep the component active
+	Enabled ManagementState = "Enabled"
+	// Unmanaged means that the operator is not taking any action related to the component
+	Unmanaged ManagementState = "Unmanaged"
+	// Disabled means that the operator is actively managing its resources and trying to remove all traces of the component
+	Disabled ManagementState = "Disabled"
+)
+
+type OperatorSpec struct {
+	// managementState indicates whether and how the operator should manage the component
+	ManagementState ManagementState `json:"managementState"`
+
+	// imagePullSpec is the image to use for the component.
+	ImagePullSpec string `json:"imagePullSpec"`
+
+	// version is the desired state in major.minor.micro-patch.  Usually patch is ignored.
+	Version string `json:"version"`
+
+	// logging contains glog parameters for the component pods.  It's always a command line arg for the moment
+	Logging LoggingConfig `json:"logging,omitempty"`
+}
+
+type LoggingConfig struct {
+	// level is passed to glog.
+	Level int64 `json:"level"`
+
+	// vmodule is passed to glog.
+	Vmodule string `json:"vmodule"`
+}
+
+type ConditionStatus string
+
+const (
+	ConditionTrue    ConditionStatus = "True"
+	ConditionFalse   ConditionStatus = "False"
+	ConditionUnknown ConditionStatus = "Unknown"
+
+	OperatorStatusTypeAvailable = "Available"
```

Comment:
Do we need a state to reflect the ManagementStatus? Eg. "Disabled: True"

Reply:
> Do we need a state to reflect the ManagementStatus? Eg. "Disabled: True"

I put it directly in the status.